### PR TITLE
PIP-1328-allow-gzipped-text-files-in-references

### PR DESCRIPTION
**Pull request by ottojolanki**

### CODE REVIEW

File: wdl/workflows/mixed/unpack.wdl [hunk 1]
```diff
@@ -65,6 +66,30 @@ workflow unpack {
...
         }
     }
+    
+    if (defined(packed_references.bias_model_gz)) {
+        call pigz.pigz as unpack_bias_model_gz_to_file {
+            input:
+                input_file=select_first([
+                    packed_references.bias_model_gz
+                ]),
+                output_filename="bias_model.txt",
```
Comment (keenangraham) [inline wdl/workflows/mixed/unpack.wdl]: Do we really want to bind the name?

Comment (ottojolanki) [reply wdl/workflows/mixed/unpack.wdl]: yeah, did the subworkflow thing, and naming is now by removing `.gz` suffix from whatever is the original filename.

File: wdl/workflows/mixed/unpack.wdl [hunk 2]
```diff
@@ -65,6 +66,26 @@ workflow unpack {
                 resources=compute.runtimes[machine_size],
         }
     }
+    
+    if (defined(packed_references.bias_model_gz)) {
+        call decompress.make_txt_from_txt_gz as unpack_bias_model_gz_to_file {
```
Comment (keenangraham) [inline wdl/workflows/mixed/unpack.wdl]: `unpack_bias_model_gz_to_file` -> `unzip_bias_model`

File: wdl/workflows/mixed/unpack.wdl [hunk 3]
```diff
@@ -65,6 +66,26 @@ workflow unpack {
...
+            input:
+                txt_gz=select_first([
+                    packed_references.bias_model_gz
+                ]),
+                resources=compute.runtimes[machine_size],
+        }
+    }
+
+    if (defined(packed_references.nuclear_chroms_gz)) {
+        call decompress.make_txt_from_txt_gz as unpack_nuclear_chroms_gz_to_file {
```
Comment (keenangraham) [inline wdl/workflows/mixed/unpack.wdl]: `unpack_nuclear_chroms_gz_to_file` -> `unzip_nuclear_chroms`

File: wdl/workflows/mixed/unpack.wdl [hunk 4]
```diff
@@ -85,9 +106,15 @@ workflow unpack {
                 unpack_hotspot2_tar_gz_to_struct.hotspot2,
                 packed_references.hotspot2
             ]),
-            nuclear_chroms: packed_references.nuclear_chroms,
+            nuclear_chroms: select_first([
+                unpack_nuclear_chroms_gz_to_file.txt,
```
Comment (keenangraham) [inline wdl/workflows/mixed/unpack.wdl]: `unpack_nuclear_chroms_gz_to_file.txt` -> `unzip_nuclear_chroms.txt`

File: wdl/workflows/mixed/unpack.wdl [hunk 5]
```diff
@@ -85,9 +106,15 @@ workflow unpack {
...
+                packed_references.nuclear_chroms
+            ]),
             narrow_peak_auto_sql: packed_references.narrow_peak_auto_sql,
-            bias_model: packed_references.bias_model
+            bias_model: select_first([
+                unpack_bias_model_gz_to_file.txt,
```
Comment (keenangraham) [inline wdl/workflows/mixed/unpack.wdl]: `unpack_bias_model_gz_to_file.txt` -> `unzip_bias_model.txt`

File: wdl/workflows/mixed/unpack.wdl [hunk 6]
```diff
@@ -65,6 +66,26 @@ workflow unpack {
                 resources=compute.runtimes[machine_size],
         }
     }
+    
+    if (defined(packed_references.bias_model_gz)) {
```
Comment (keenangraham) [inline wdl/workflows/mixed/unpack.wdl]: Should bias_model go below nuclear_chroms?

**Review by keenangraham — CHANGES_REQUESTED** [review]

**Review by keenangraham — APPROVED** [review]
Awesome.

**Pull request merged**